Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-dashboardblockbasedview', function (Y) {
+    'use strict';
+    /**
+     * Provides the Dashboard Block Based View class
+     *
+     * @module ez-dashboardblockbasedview
+     */
+    Y.namespace('eZ');
+
+    var SELECTOR_CONTENT = '.dashboard-content';
+
+    /**
+     * The dashboard view
+     *
+     * @namespace eZ
+     * @class DashboardBlockBasedView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.DashboardBlockBasedView = Y.Base.create('dashboardBlockBasedView', Y.eZ.TemplateBasedView, [], {
+        initializer: function () {
+            this.after('activeChange', function () {
+                this.after('blocksChange', this._renderBlocks, this);
+
+                this._makeBlocksActive();
+            }, this);
+        },
+
+        /**
+         * Renders the dashboard view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockBasedView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template({dashboardTitle: this.get('dashboardTitle')}));
+            this._renderBlocks();
+
+            return this;
+        },
+
+        /**
+         * Activates all the blocks to allow communication between blocks and service
+         *
+         * @method _makeBlocksActive
+         * @protected
+         */
+        _makeBlocksActive: function () {
+            var blocks = this.get('blocks');
+
+            Object.keys(blocks).forEach(function (key) {
+                blocks[key].set('active', true);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this._sortBlocks(this.get('blocks')).forEach(Y.bind(function (block) {
```

Comment:
this method does too many things. Also from a test perspective, it would be easier to move the sorting and the `addTarget` call to `addView`. So basically, this method only renders the blocks (as its name suggests).